src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -134,42 +133,54 @@ private bool TryCreateAutoResolveBindingTemplate(PropertyInfo propInfo, INameRes
         {
             template = null;
 
-            AutoResolveAttribute attr = propInfo.GetCustomAttribute<AutoResolveAttribute>();
+            string resolvedValue = null;
+            if (!TryAutoResolveValue(_source, propInfo, nameResolver, out resolvedValue))
```

Comment:
All the changes here are me making the code more unit testable - pushing logic out into TryAutoResolveValue static